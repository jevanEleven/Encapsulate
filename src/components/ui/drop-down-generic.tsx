/**
 * @module Component for handling general dropdown needs.
 * @param String stateObject: linked to the dropdown storing its value.
 * @param Function updateState: setState function linked to the dropdown updating its value.
 * @param Array<String> options: array of strings giving the dropdown's potential options.
 * @param String Label: Label to give to the dropdown.
 * @returns
 */

/**
 *
 *
 */
interface DropDownProps {
    stateObject: string;
    updateState: (newValue: string) => void;
    options: Array<string>;
    label: string;
}

export default function GenericDropDown(props: DropDownProps) {
    const { stateObject, updateState, options, label } = props;

    return (
        <div className="flex items-center">
            <span className="mr-1 text-sm">{label}</span>
            <select
                value={stateObject}
                onChange={(e) => {
                    updateState(e.target.value);
                }}
                className="border border-gray-300 rounded px-1 py-1 text-sm"
            >
                {options.map((option, index) => {
                    return (
                        <option key={index} value={option.toLowerCase()}>
                            {option}
                        </option>
                    );
                })}
            </select>
        </div>
    );
}
