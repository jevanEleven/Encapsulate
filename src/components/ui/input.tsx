import * as React from "react";

import { cn } from "@/lib/utils";

export interface InputProps
    extends React.InputHTMLAttributes<HTMLInputElement> {}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
    ({ className, type, ...props }, ref) => {
        const inputProps =
            type === "checkbox"
                ? {
                      ...props,
                      checked: !!props.checked,
                  }
                : {
                      ...props,
                      value: props.value,
                  };

        return (
            <input
                type={type}
                className={cn(
                    "flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50",
                    className
                )}
                ref={ref}
                {...inputProps}
            />
        );
    }
);
Input.displayName = "Input";

interface MultiFileInputProps
    extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "value"> {}

const MultiFileInput = React.forwardRef<HTMLInputElement, MultiFileInputProps>(
    ({ className, ...props }, ref) => {
        return (
            <input
                type="file"
                multiple
                className={cn(
                    "flex w-full rounded-md border border-input bg-transparent py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50",
                    className
                )}
                ref={ref}
                {...props}
            />
        );
    }
);
MultiFileInput.displayName = "MultiFileInput";

export interface TextAreaProps
    extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}

const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
    ({ className, ...props }, ref) => {
        return (
            <textarea
                className={cn(
                    "flex w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50",
                    className
                )}
                ref={ref}
                {...props}
            />
        );
    }
);

TextArea.displayName = "TextArea";

export { Input, MultiFileInput, TextArea };
