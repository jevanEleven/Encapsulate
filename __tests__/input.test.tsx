import { render } from "@testing-library/react";
import React from "react";
import { Input, MultiFileInput, TextArea } from "@/components/ui/input"

describe("Form Inputs", () => {
  it("renders a default Input component", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  it("renders a default MultiFileInput component", () => {
    const { container } = render(<MultiFileInput />);
    expect(container).toMatchSnapshot();
  });

  it("renders a default TextArea component", () => {
    const { container } = render(<TextArea />);
    expect(container).toMatchSnapshot();
  });
});