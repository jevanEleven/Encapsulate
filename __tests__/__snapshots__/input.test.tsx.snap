// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Inputs renders a default Input component 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
    value=""
  />
</div>
`;

exports[`Form Inputs renders a default MultiFileInput component 1`] = `
<div>
  <input
    class="flex w-full rounded-md border border-input bg-transparent py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
    multiple=""
    type="file"
  />
</div>
`;

exports[`Form Inputs renders a default TextArea component 1`] = `
<div>
  <textarea
    class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
  />
</div>
`;
